src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportFromFileService.java

```diff
@@ -85,11 +83,29 @@ public ProgrammingExercise importProgrammingExerciseFromFile(ProgrammingExercise
         if (Boolean.TRUE.equals(programmingExerciseForImport.isStaticCodeAnalysisEnabled())) {
             staticCodeAnalysisService.createDefaultCategories(importedProgrammingExercise);
         }
+        copyEmbeddedFiles(exerciseFilePath.toAbsolutePath().getParent().resolve(FileNameUtils.getBaseName(exerciseFilePath.toString())));
         importRepositoriesFromFile(importedProgrammingExercise, importExerciseDir, oldShortName);
         importedProgrammingExercise.setCourse(course);
+        fileService.scheduleForDirectoryDeletion(importExerciseDir, 5);
         return importedProgrammingExercise;
     }
 
+    /**
+     * copy embedded files from the extracted zip file to the markdown folder, so they can be used in the problem statement
+     **/
+    private void copyEmbeddedFiles(Path importExerciseDir) throws IOException {
+        // the conversion to a string is needed because otherwise the file extension is not included
+        Predicate<Path> embeddedFile = file -> file.getFileName().toString().endsWith(".jpg") || file.getFileName().toString().endsWith(".jpeg")
+                || file.getFileName().toString().endsWith(".png") || file.getFileName().toString().endsWith(".gif") || file.getFileName().toString().endsWith(".svg")
+                || file.getFileName().toString().endsWith(".pdf");
+        try (var embeddedFiles = Files.list(importExerciseDir).filter(embeddedFile)) {
```

Comment:
In the export you place all images in a subdirectory `"programming-exercise-material"`. Could you only walk over that directory and copy its content without additional filter?

Reply:
This is the subdirectory that is used for all the export material, it contains more than the images. (exercise details json file, problem statement markdown file, zipped repository), but I think I could place it in an additional subdirectory and hence get rid of the filter